src/Storages/ObjectStorage/DataLakes/Iceberg/IcebergMetadataFilesCache.h

```diff
@@ -0,0 +1,190 @@
+#pragma once
+#include "config.h"
+
+#if USE_AVRO
+
+#include <Core/Settings.h>
+#include <Common/CacheBase.h>
+#include <Common/HashTable/Hash.h>
+#include <Common/logger_useful.h>
+#include <Storages/ObjectStorage/DataLakes/Iceberg/Snapshot.h>
+#include <Storages/ObjectStorage/DataLakes/Iceberg/ManifestFile.h>
+#include <Storages/ObjectStorage/StorageObjectStorage.h>
+
+namespace ProfileEvents
+{
+    extern const Event IcebergMetadataFilesCacheMisses;
+    extern const Event IcebergMetadataFilesCacheHits;
+    extern const Event IcebergMetadataFilesCacheWeightLost;
+}
+
+namespace CurrentMetrics
+{
+    extern const Metric IcebergMetadataFilesCacheSize;
+}
+
+namespace DB
+{
+
+/// Manifest list contains only a list of manifest files, so we don't have to store the deserialized manifest_list content.
+/// In ManifestListCacheCell we store a list of manifest files' data path and sequence number,
+/// so we can get manifest files from cache or read and construct them.
+struct ManifestListCacheCell
+{
+    struct ManifestFileEntry
+    {
+        String data_path;
+        Int64 added_sequence_number;
+    };
+    using CachedManifestList = std::list<ManifestFileEntry>;
+    CachedManifestList cached_manifest_files;
+    size_t getMemoryBytes() const
+    {
+        size_t total_size = sizeof(std::list<ManifestFileEntry>);
+        for (const auto & entry: cached_manifest_files)
+        {
+            total_size += sizeof(ManifestFileEntry) + entry.data_path.capacity();
+        }
+        return total_size + 200;
```

Comment:
Better to move 200 to a constexpr defined variable with some self-explanatory naming